### PR TITLE
Toast: new experimental Toast redesign

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

Toast: implementation new Toast design behind experiment

| BEFORE      | AFTER |
| ----------- | ----------- |
| Screen Shot 2022-11-18 at 11 43 10 AM |  Screen Shot 2022-11-18 at 11 42 42 AM |
| Screen Shot 2022-11-18 at 11 44 59 AM | Screen Shot 2022-11-18 at 11 44 46 AM |

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4028)
- [TDD](https://paper.dropbox.com/doc/Gestalt-TDD-Toast--BqZRSpB7ziwSUFAwTIGjaPx2Ag-EC6LAr0uO808c6GrjH9FO)
- [Figma](http://localhost:8888/web/toast#Error)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/docs-components/contexts/DocsExperimentProvider.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { ExperimentProvider } from 'gestalt';
  * - Unless you want the experimental behavior live on the docs for everyone, REMOVE YOUR EXPERIMENT HERE before merging your PR!
  * */
 
-const enabledExperiments = [];
+const enabledExperiments = ['web_gestalt_redesigned_toast', 'mweb_gestalt_redesigned_toast'];
```
Comment (dangerismycat) [inline docs/docs-components/contexts/DocsExperimentProvider.js:11]: Same note here about not leaving these in until we provide the user a way to toggle them off.

File: docs/pages/web/toast.js [hunk 2]
```diff
@@ -15,25 +16,39 @@ export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen
...
-      src="https://i.ibb.co/Lx54BCT/stock1.jpg"
-    />
-  }
-/>
-        `}
+  <Toast
+    primaryAction={{ accessibilityLabel: 'Test', label: 'Undo', size: 'lg' }}
+    text={
+      <Text inline> Saved to
+        <Link inline target="blank" href="https://www.pinterest.com/search/pins/?q=home%20decor">
```
Comment (dangerismycat) [inline docs/pages/web/toast.js:23]: Can we use `href="#"` so the user doesn't accidentally navigate away from the page?

File: packages/gestalt/src/Toast.js [hunk 3]
```diff
@@ -66,6 +67,24 @@ export default function Toast({
   variant = 'default',
   _dangerouslySetPrimaryAction,
 }: Props): Node {
+  /*
+   *
+   * EXPERIMENT LOGIC START
```
Comment (dangerismycat) [inline packages/gestalt/src/Toast.js:72]: this seems excessive for seven lines of code?

File: packages/gestalt/src/Toast.js [hunk 4]
```diff
@@ -66,6 +67,24 @@ export default function Toast({
...
+   *
+   */
+  const inToastExp = useInExperiment({
+    webExperimentName: 'web_gestalt_redesigned_toast',
+    mwebExperimentName: 'mweb_gestalt_redesigned_toast',
+  });
+
+  const EXP_SIZE_THUMBNAIL = 32;
```
Comment (dangerismycat) [inline packages/gestalt/src/Toast.js:80]: These should be defined outside of the function

**Review by dangerismycat — APPROVED** [review]
nit: can you crop the before/after images so they're easier to see? The additional whitespace on either side isn't helpful.

File: docs/docs-components/contexts/DocsExperimentProvider.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/docs-components/contexts/DocsExperimentProvider.js]: Moved to new logic

File: docs/pages/web/toast.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/pages/web/toast.js:26]: DONE

File: packages/gestalt/src/Toast.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Toast.js:72]: DONE

File: packages/gestalt/src/Toast.js (hunk 4 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Toast.js:80]: DONE

**Pull request merged**